src/firmware/main/world/chicker.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+/**
+ * This struct represents a robot chicker (kicker and chipper) from the perspective of
+ * firmware
+ */
+struct Chicker;
+typedef struct Chicker Chicker;
+
+/**
+ * Create a chicker with the given functions for interacting with it
+ *
+ * @param kick A function that we can call to kick with the chicker with a given speed,
+ *             in meters per second
+ * @param chip A function that we can call to chip a given distance, in meters, with the
```

Comment:
Nit:
Maybe we should specify the chip distance is to the frst contact point with the field (vs chip + roll distance or something)?